File: perl/reader.pl
#!/usr/bin/env perl
# Reader.pl
# Read in a specified file or dump out the specified number of records.
use strict;
use warnings;
use Parser;

my ($command, $param) = @ARGV;

if (not defined $command) {
  print "Usage:\n\t$0 parse <filename>\n\t$0 fetch <num records>\n";
  exit(1);
}

if ($command =~ /parse/) {
    if (not defined $param) {
        print "Usage:\n\t$0 parse <filename>\n\t$0 fetch <num records>\n";
        exit(1);
    }
    my $db = new WeblogDB("test.db");
    my $p = new Parser($db);
    print "Reading file: $param\n";
    open (SINK, "<$param");
    while (my $line = <SINK>) {
      $p->parseWeblog($line);
    }
    close SINK;
    print "Processed " . $p->count . " logs.\n";
    $p->toDatabase();
    print "Wrote to database ok.\n";
    exit(0);
} elsif ($command =~ /fetch/) {
    my $db = new WeblogDB("test.db");
    $param = 50 if (not defined $param);
    print "Fetching $param records from database...\n";
    my @logs = $db->fetch_all($param);
    foreach my $log (@logs) {
        print $log->to_string . "\n";
    }
} else {
    print "Usage:\n\t$0 parse <filename>\n\t$0 fetch <num records>\n";
    exit(1);
}

File: ruby/reader.rb
#!/usr/bin/env ruby
# Reader.rb
# Read in a specified file or dump out the specified number of records
# @author FoxBoxPDX
require_relative 'weblogmodule'

command = ARGV.shift
param = ARGV.shift

if command.nil?
  puts "Usage:\n\t#{$0} parse <filename>\n\t#{$0} fetch <num records>\n\t#{$0} create table"
  exit(1)
end

case command
when 'parse'
  if param.nil?
    puts "Usage:\n\t#{$0} parse <filename>\n\t#{$0} fetch <num records>\n\t#{$0} create table"
    exit(1)
  end
  db = WeblogDB.new("test.db")
  p = Parser.new(db)
  puts "Reading file: #{param}"
  f = File.open(param, 'r')
  f.each_line { |line| p.parse_weblog(line) }
  f.close
  puts "Processed #{p.count} logs."
  p.to_database
  puts "Wrote to database ok."
  exit(0)
when 'fetch'
  db = WeblogDB.new("test.db")
  param = 50 if param.nil?
  puts "Fetching #{param} records from database..."
  logs = db.fetch_all(param)
  logs.each { |log| puts log }
  exit(0)
when 'create'
  db = WeblogDB.new("test.db")
  db.create_tables
  puts "Created Tables"
else
  puts "Usage:\n\t#{$0} parse <filename>\n\t#{$0} fetch <num records>\n\t#{$0} create table"
  exit(1)
end
